src/util/side_metadata/helpers.rs

```diff
@@ -72,43 +77,17 @@ pub(super) fn try_mmap_contiguous_metadata_space(
     let mmap_size =
         address_to_meta_address(*spec, start + size).align_up(BYTES_IN_PAGE) - mmap_start;
     if mmap_size > 0 {
-        // FIXME - This assumes that we never mmap a metadata page twice.
-        // While this never happens in our current use-cases where the minimum data mmap size is a chunk and the metadata ratio is larger than 1/64, it could happen if (min_data_mmap_size * metadata_ratio) is smaller than a page.
-        // E.g. the current implementation detects such a case as an overlap and returns false.
         if !no_reserve {
-            try_mmap_metadata(mmap_start, mmap_size)
+            MMAPPER.ensure_mapped(mmap_start, mmap_size >> LOG_BYTES_IN_PAGE)
```

Comment:
This works ok if `mmap_size` is always a multiple of page size.
Otherwise we may use not-mmaped memory

Reply:
I think the calculation above makes sure `mmap_size` is always page aligned.

Reply:
Oh sorry you're right